src/btree/bt_split.c

```diff
@@ -1450,7 +1451,8 @@ __split_multi_inmem(WT_SESSION_IMPL *session, WT_PAGE *orig, WT_MULTI *multi, WT
          * them back to their original update chains. Truncate before we restore them to ensure the
          * size of the page is correct.
          */
-        if (supd->onpage_upd != NULL && !F_ISSET(S2C(session), WT_CONN_IN_MEMORY)) {
+        if (supd->onpage_upd != NULL && !F_ISSET(S2C(session), WT_CONN_IN_MEMORY) &&
+          !F_ISSET(S2BT(session), WT_BTREE_IN_MEMORY)) {
```

Comment:
Nitpick, we're now doing a lot of these `WT_CONN_IN_MEMORY && WT_BTREE_IN_MEMORY` checks paired together. Should we have a macro to combine them?